amazon_transcribe/serialize.py

```diff
@@ -25,62 +24,93 @@
     BaseEvent,
 )
 
-HEADER_VALUE = Union[int, None, str]
 
+class TranscribeStreamingSerializer:
+    """Convert StartStreamTranscriptionRequest into a
+    Request object for streaming to the Transcribe service.
+    """
 
-class Serializer:
-    def __init__(self):
-        raise NotImplementedError("Serializer")
+    def _serialize_header(
+        self, header: str, value: Any, prefix: str = "x-amzn-transcribe-",
+    ) -> Dict:
```

Comment:
In this case would it make better sense to include something like 
```
HEADER_VALUE = Union[Optional[str], Optional[int], Optional[bool]] 
```
and use that in place of `Any`

Reply:
Yes, something like that is definitely reasonable. If for nothing more than saying "hey when we wrote this these were the types we had considered others may not be defined". If we had more information and strong guarantees on what exactly we intend to do with the value in the body it makes sense to use a type that more so represents the interface than the particular object type. For example if we had a function we could say it takes `Union[list, dict, tuple]`, but if all we used it for was iterating and that's all we cared about perhaps `Iterable` would be the better type since it more accurately reflects what we're expecting of the value passed in. In this specific case our only expectation is that the value is str-able which I don't think there's a more reasonable type for than `Any` since anything can be cast as a str.

As an aside: `Optional[Foo]` is really just a fancy way of saying `Union[None, Foo]`. So in your example it would simplify to `Union[None, str, None, int, None, bool]`, at which point you might as well just say `Union[None, str, int, bool]` or perhaps `Optional[Union[str, int, bool]]`.